cupyx/scipy/ndimage/measurements.py

```diff
@@ -200,3 +202,175 @@ def _kernel_finalize():
         y[i] = j + 1;
         ''',
         'cupyx_nd_label_finalize')
+
+
+def _variance_kernel():
+    input_arguments = 'T image, R label, raw X index, int32 size, raw Z mean'
+    output_arguments = 'raw Z ret'
+    _kernel_template = """
+    for (ptrdiff_t j = 0; j < size; j++) {
+      if (label == index[j]) {
+        atomicAdd(&ret[j], (image - mean[j]) * (image - mean[j]));
+        break;
+      }
+    }
+    """
+    return cupy.ElementwiseKernel(input_arguments, output_arguments,
+                                  _kernel_template)
+
+
+def _stats(input, labels=None, index=None, mean=False):
+
+    input_arguments = 'T image, R label, raw X index, int32 size'
+    _kernel_template = string.Template("""
```

Comment:
Isolate the kernel creation and memoize it

Reply:
Done